types/wlr_surface.c

```diff
@@ -325,6 +325,10 @@ static void wlr_surface_apply_damage(struct wlr_surface *surface,
 					surface->current->buffer)) {
 			wlr_texture_upload_drm(surface->texture, surface->current->buffer);
 			goto release;
+		} else if (wlr_dmabuf_buffer_from_buffer_resource(
+					   surface->current->buffer)) {
```

Comment:
`wlr_dmabuf_buffer_from_buffer_resource` aborts if the resource isn't a dmabuf buffer or if the buffer is NULL.

You seem to assume here (and in other places too, grep for `wlr_dmabuf_buffer_from_buffer_resource`) that it returns NULL instead of aborting. Maybe a `wlr_dmabuf_resource_is_buffer` could be used instead?